### PR TITLE
Jsoo stubs for mtime.clock.os

### PR DESCRIPTION
**Pull request by hhugo**
Sister PR of https://github.com/dbuenzli/ptime/pull/25

### CODE REVIEW

**Comment by dbuenzli** [comment]
@hhugo is on fire. Let me first deal with your `brr` work asap…

**Comment by hhugo** [comment]
Do you have more libraries with the same design you want me to look at ?

**Comment by dbuenzli** [comment]
Thanks for the offer ! I have a few other things with `jsoo` backends but I don't think this technique can be applied there. Basically this works well for C bindings.

File: pkg/META [hunk 1]
```diff
@@ -30,6 +30,7 @@ package "clock" (
     archive(native) = "mtime_clock.cmxa"
     plugin(byte) = "mtime_clock.cma"
     plugin(native) = "mtime_clock.cmxs"
+    linkopts(javascript) = "+mtime/os/runtime.js"
```
Comment (dbuenzli) [inline pkg/META:33]: @hhugo in ptime you used `.` as separators ( "+ptime.clock.os") rather than `/`, does `jsoo` support both ?

Comment (hhugo) [reply pkg/META:33]: jsoo take the part between `+` and the first`/` to figure out the package and ask findlib for the directory.
The part after the first `/` is appended to that directory

Comment (dbuenzli) [reply pkg/META:33]: Ok thanks understood.

Comment (hhugo) [reply pkg/META:33]: The version with the `.` is probably more accurate. `"+mtime.clock.os/runtime.js"`

Comment (dbuenzli) [reply pkg/META:33]: Okay don't bother, I'll change that in the reshuffle.

File: src-os/runtime.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  test_perf_now(globalThis.perf_hooks)
+if(performance_now === undefined && typeof require == "function"){
+  var ph = require("perf_hooks");
+  test_perf_now(ph);
+}
+if(performance_now === undefined)
+  performance_now = (function () { caml_raise_sys_error("performance.now () is not available"); })
+function mtime_clock_now(){
+  var now_ms = performance_now ();
+  var now_ns = caml_int64_of_float(now_ms * 1e6);
```
Comment (dbuenzli) [inline src-os/runtime.js:26]: The original code had a more elaborate [conversion procedure](https://github.com/dbuenzli/mtime/blob/f472ffdf8bf2d3f941980d63073025df847c1173/src-jsoo/mtime_clock.ml#L34-L60). Did you find it unnecessary ?

**Comment by dbuenzli** [comment]
Your patch is in as f472ffdf8bf2d3f941980. Thanks.

File: src-os/runtime.js (hunk 2 above)
Comment (dbuenzli) [reply src-os/runtime.js:26]: So in a2ab7d45debabca0a86fdb I reinstated the old procedure. If we do the conversion that way you suggested here then, unless my logic is broken, I think we could already roll over after: 
```
# 2. ** 53. /. 1_000_000_000. /. (24. *. 3600. *. 365.25) ;;
- : float = 0.285420920942688672
```
Julian years, that is ~102 days. 

While I don't see a browser tab running for so long, I can see a potential node program do (but who does that when you have native compilation available :-).

Comment (dbuenzli) [reply src-os/runtime.js:26]: (Actually it's not: roll over it's: becoming imprecise).

Comment (hhugo) [reply src-os/runtime.js:26]: I didn't spend much time on this. I don't have any opinion on the procedure used.

Comment (dbuenzli) [reply src-os/runtime.js:26]: > I didn't spend much time on this.

I can see that ;-)

In fact your code doesn't work at all in the browser, you can't call `performance.now` directly as a function. If you checkout the [previous code](https://github.com/dbuenzli/mtime/blob/a2ab7d45debabca0a86fdbb73a3dde742c935a99/src-jsoo/mtime_clock.ml#L15-L32) it was a method call. e3b263b4bbdb179f605ac fixes it.

**Pull request closed without merging**